openllm-core/src/openllm_core/utils/__init__.py

```diff
@@ -71,7 +71,12 @@ def lenient_issubclass(cls: t.Any, class_or_tuple: type[t.Any] | tuple[type[t.An
 
 def ensure_exec_coro(coro: t.Coroutine[t.Any, t.Any, t.Any]) -> t.Any:
   loop = asyncio.get_event_loop()
-  if loop.is_running(): return asyncio.run_coroutine_threadsafe(coro, loop).result()
+  if loop.is_running():
+    # in nootbook run_coroutine_threadsafe will got stuck for some reason (https://github.com/jupyter/notebook/issues/5261), so use asyncio.run instead
```

Comment:
This has to do with jupyter notebook has a running event loop

Reply:
yes.